app/models/card.rb

```diff
@@ -1,16 +1,26 @@
 class Card < ActiveRecord::Base
   belongs_to :user
+  belongs_to :block
   validates :user_id, presence: true
   before_validation :set_review_date, on: :create
   validate :texts_are_not_equal
   validates :original_text, :translated_text, :review_date,
             presence: { message: 'Необходимо заполнить поле.' }
   validates :user_id, presence: { message: 'Ошибка ассоциации.' }
-
-  scope :pending, -> { where('review_date <= ?', Time.now).order('RANDOM()') }
+  validates :block_id,
+            presence: { message: 'Выберите колоду из выпадающего списка.' }
 
   mount_uploader :image, CardImageUploader
 
+  def self.pending(current_block)
+    if current_block
+      where('review_date <= ? AND block_id = ?',
+            Time.now, current_block).order('RANDOM()')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.